samples/memfs/memfs.go

```diff
@@ -615,6 +616,24 @@ func (fs *memFS) OpenFile(
 	if !inode.isFile() {
 		panic("Found non-file.")
 	}
+	// What perms do we need?
```

Comment:
I had trouble understanding what you were doing here at first. Could you make
the code read something like:


    // Check that the open mode is legal according to the file
    // permissions.
    var permissionBits int
    switch (...) {
        ...
    }
    
    if (...) {
      err = EACCESS
      return
    }

Reply:
we should not be checking attributes, since they can change behind our back.